guides/integrate-nextauth.md

```diff
@@ -0,0 +1,327 @@
+    let passkeys = await embedded.getPasskeys();
+    let promptText = passkeys.map((passkey, index) => {
+      return `${index}: ${passkey.identity.username}`;
+    }).join("\n");
+    let selectedIndex = parseInt(prompt(promptText, "index")!!);
+    if (selectedIndex >= 0 && selectedIndex < passkeys.length) {
+      let selectedId = passkeys[selectedIndex].id;
+      // Perform authentication using selected id
+      let result = await embedded.authenticate(url, selectedId);
+      return Promise.resolve(result.redirectURL);
+    } else {
+      // This will fail in core as it won't match to any id
+      return Promise.resolve("unknown_id");
+    }
+  }
+
+  return (
+    <div
+      style={{
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+        height: '100vh',
+      }}
+    >
+      <div className="container">
+        <div className="row">
+          <div className="d-flex justify-content-center">
+            <div className="spinner-border" role="status">
+              <span className="sr-only"></span>
+            </div>
+          </div>
+        </div>
+        <div className="row">
+          {
+            biAuthenticateResult.length > 0 &&
+            <div className="row row-cols-1 row-cols-md-1 mt-3">
+              <div className="col">
+                <code>
+                  {JSON.stringify(biAuthenticateResult, null, 2)}
+                </code>
+              </div>
+            </div>
+          }
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default BIAuthenticate;
+```
+
+What's happening here?
+
+1. The `useEffect` is only called once on page load. In this function, we initialize the Beyond Identity SDK and use `embedded.isAuthenticateUrl` to check if the current page that was redirected to is in fact a valid `bi-authenticate` URL.
+2. If the URL is valid, we pull the URL using `window.location.href` and pass that directly into `biAuthenticate` in step 3.
+3. `biAuthenticate` calls `embedded.authenticate` with a valid `bi-authenticate` URL. This function performs a challenge/response against a passkey bound to your browser. Note that the callback in `embedded.authenticate` contains logic in order to prompt a user to select a passkey if there is more than one.
+4. Finally, the response of `embedded.authenticate` contains a `redirectURL`. Follow this redirectURL to complete the OAuth/OIDC flow.
+
+### Token Exchange
+
+Calling the token endpoint is the second step in the authorization flow and usually happens in your backend if your application's Client Type is `Confidential`. Make sure to a call the [authorization endpoint](/docs/v1/workflows/authentication#1-craft-authorization-url) first to retrieve an authorization code.
+
+If your application is using the [NextAuth](https://next-auth.js.org) provider (see the Javascript Authorization example using Automatic Invocation Type), you will not need to complete authentication with a token exchange.
```

Comment:
I'm not sure if we need this section either. We already mention that the provider will go through an OAuth/OIDC that will result in fetching an id token that will log you in to the example app in "Configuring NextAuth"

Reply:
True. Done.